### PR TITLE
Fix for tabular metadata uploads that only contain one row

### LINKED ISSUE #1463: Tabular files with a single row don't show up properly in data_sets 1  file browser
* Specific code commit: fad81a20e6ce1334cde6c3350a1bf89fa4b82a81
* Version of the web browser and OS: `Chrome Version 53.0.2785.143 (64-bit)`
* Environment where the error occurred (Vagrant VM and site conf mode or AWS instance): https://test.stemcellcommons.org

### Steps to reproduce
1. Upload this comma delimited file (or any other tabular file referencing a single data file) into Refinery: [sample-comma.txt](https://github.com/parklab/refinery-platform/files/523334/sample-comma.txt)

2. View the File Browser at `https://test.stemcellcommons.org/data_sets/<uuid>`:
  * See no files displayed (no console errors): screen shot 2016-10-11 at 9 43 28 pm

3. View the File Browser at `https://test.stemcellcommons.org/data_sets2/<uuid>`:
  * See the proper behavior: 
screen shot 2016-10-11 at 9 43 45 pm

### Observed behavior
* I haven't exhausted all edge cases but it seems that tabular files that reference a single data file, regardless of delimiter type,  aren't displayed properly in the `data_sets` file broswer, but are displayed properly in `data_sets2` file browser

### Expected behavior
* For the `data_sets` file browser to be able to handle tabular files referencing a single data file.

### PR DESCRIPTION
**Pull request by scottx611x**
Fixes #1463

### CODE REVIEW

**Comment by jkmarx** [comment]
@scottx611x @ngehlenborg 
Did you decide if we should handle the matrix page? Provenance also appears broken for a single row data set. Is this expected? I'm using data set 15826: ChipSeq of MLL1 in mouse leukemic blast cells

screen shot 2016-10-19 at 4 36 23 pm

File: refinery/static/source/js/refinery/solr/solr_response.js [hunk 1]
```diff
@@ -60,94 +60,96 @@ SolrResponse.prototype.getDocumentList = function() {
...
 	for ( var i = 0; i < response.response.docs.length; ++i ) {
 		documentList.push( response.response.docs[i] );
     }
-    
+
     return documentList;
 }
 
 
 SolrResponse.prototype._processPivotCounts = function ( response ) {
```
Comment (jkmarx) [inline refinery/static/source/js/refinery/solr/solr_response.js]: self._pivotCounts = self._processPivotCounts(response)
The default value for self._pivotCount = {}
I'd recommend setting an else with {}, even though no extra errors seem to be thrown.

File: refinery/core/views.py [hunk 2]
```diff
@@ -715,13 +715,18 @@ def solr_core_search(request):
...
     # core format is <name_of_core>
     # query.GET is a querydict containing all parts of the query
-    # TODO: handle runtime errors when making GET request
-
     url = settings.REFINERY_SOLR_BASE_URL + core + "/select"
     data = request.GET.urlencode()
     try:
         full_response = requests.get(url, params=data)
-        full_response.raise_for_status()
+        # Solr sends back an additional 400 here in the data_sets 1 filebrowser
```
Comment (jkmarx) [inline refinery/core/views.py]: API will stick around. It's used as a UI common services solr.js, which is used by multiple features.
